src/sentry/search/events/types.py

```diff
@@ -105,6 +106,19 @@ def parse_stats_period(self) -> None:
 
             self.start = get_datetime_from_stats_period(self.stats_period, self.end)
 
+    @property
+    def start_date(self) -> datetime:
+        # This and end_date are helper functions so callers don't have to check if either are defined for typing
+        if self.start is None:
+            raise InvalidSearchQuery("start is required")
+        return self.start
```

Comment:
I'm not hugely in love with this, might revisit it later to see if i can make it cleaner